lib/super_good/solidus_taxjar/reporting.rb

```diff
@@ -5,6 +5,11 @@ def initialize(api: SuperGood::SolidusTaxjar.api)
         @api = api
       end
 
+      def refund_and_create_new_transaction(order)
```

Comment:
I think you are right. Perhaps more sophisticated error-handling and/or transaction wrapping could be a future PR.

In the meantime, maybe it is even enough to do `@api.create_refund_transaction_for(order) && @api.create_transaction_for(order)` to stop the second call from happening if the first fails.

Reply:
I like Ben's suggestion to not create the updated transaction if creating the refund fails. That should handle at least one of the two possible failure cases here!

If the create transaction fails we could [delete the refund](https://developers.taxjar.com/api/reference/#delete-delete-a-refund-transaction), but that still leaves us in an inconsistent state, where Solidus and TaxJar have different representations of the order. Since this is performed in a job we could delete the refund transaction and then rely on someone to retry the job manually. We could also start tracking these kind of failures and surface them in the reporting UI in admin, but that would be a bit more work to build out.

I am curious if it's worth maybe still creating a `taxjar_order_transaction` or `taxjar_refund_transaction` record and saving the failure response code and message on it 🤔

Reply:
> In the meantime, maybe it is even enough to do @api.create_refund_transaction_for(order) && @api.create_transaction_for(order) to stop the second call from happening if the first fails.

Yes, this is great!

@forkata @benjaminwil maybe the question of what to do in a partial failure we could bring up with Ryan in our meeting with him today. I do expect part of the answer will be logging of some sort, but "do we/how do we" surface this in the admin is an good question to ask.